### PR TITLE
Dpl 543 runs page bug

### PR DESCRIPTION
**Pull request by seenanair**
- Mapping value field in custom component
- Fall through attribute binding
-  Added unit test cases

### CODE REVIEW

**Review by stevieing — CHANGES_REQUESTED** [review]
All seems good to me. Might be worth adding back in some comments about what kind of attributes can be passed along with a bit about how it works?

File: src/components/shared/TractionInput.vue [hunk 1]
```diff
@@ -3,14 +3,9 @@
     <label v-if="title">{{ title }}</label>
     <input
       :id="id"
-      :value="content"
+      v-bind="$attrs"
```
Comment (BenTopping) [inline src/components/shared/TractionInput.vue:6]: Does this work by spreading/binding all passed in props to this component? If so does that mean we still need to explicitly declare :value ?

File: tests/unit/components/shared/TractionInput.spec.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        test: '',
+      },
+    })
+    const textInput = comp.find('input')
+    textInput.element.value = 'Test value'
+    textInput.trigger('input')
+    expect(comp.vm.test).toBe('Test value')
+  })
+
+  it('updates the input when changing the v-model value externally,', async () => {
```
Comment (BenTopping) [inline tests/unit/components/shared/TractionInput.spec.js:60]: Nice test!

Comment (BenTopping) [reply tests/unit/components/shared/TractionInput.spec.js:60]: Is the <div> surrounding <traction-input> needed?

**Comment by seenanair** [comment]
> All seems good to me. Might be worth adding back in some comments about what kind of attributes can be passed along with a bit about how it works?

Sure, will do

File: src/components/shared/TractionInput.vue (hunk 1 above)
Comment (seenanair) [reply src/components/shared/TractionInput.vue:6]: From the documentation I read, it says we need explicit binding for value field as this is a custom component wrapper.

File: tests/unit/components/shared/TractionInput.spec.js (hunk 2 above)
Comment (seenanair) [reply tests/unit/components/shared/TractionInput.spec.js:60]: Removed 👍

**Comment by seenanair** [comment]
> All seems good to me. Might be worth adding back in some comments about what kind of attributes can be passed along with a bit about how it works?

@stevieing  Comments added

File: src/components/shared/TractionInput.vue (hunk 1 above)
Comment (BenTopping) [reply src/components/shared/TractionInput.vue:17]: 👍 Do you have a link to the doc?

Comment (seenanair) [reply src/components/shared/TractionInput.vue:17]: https://v2.vuejs.org/v2/guide/components.html#Using-v-model-on-Components
https://vuejs.org/guide/components/attrs.html
Also, if you search for Creating an Input component using Vue, you can find similar documentation - like https://dev.to/abhisheksarmah/creating-an-input-component-using-vue-and-tailwindcss-3a6b

**Comment by seenanair** [comment]
@stevieing @BenTopping  I think I have made the changes you suggested
- Documentation added
- test case update
- removed id field prop as it will mapped as part of $attrs

**Review by stevieing — APPROVED** [review]

**Pull request merged**